cpp/src/rolling/detail/rolling_utils.cuh

```diff
@@ -17,14 +17,28 @@
 #pragma once
 
 #include <cudf/detail/iterator.cuh>
+#include <cudf/detail/rolling.hpp>
 #include <cudf/types.hpp>
 
 #include <cuda/functional>
+#include <cuda/std/type_traits>
 
 namespace CUDF_EXPORT cudf {
 
 namespace detail::rolling {
 
+/**
+ * @brief Information about group bounds of the current row's group.
+ */
+struct range_group_info {
+  size_type group_start;
+  size_type group_end;
+  size_type null_start;
+  size_type null_end;
+  size_type non_null_start;
+  size_type non_null_end;
+};
+
```

Comment:
Added this struct and moved the grouped/ungrouped with nulls structs here from `range_utils.cuh`.